app/services/record_declarations/base.rb

```diff
@@ -1,12 +1,37 @@
 # frozen_string_literal: true
 
-require "json_schema/validate_body_against_schema"
-
 module RecordDeclarations
   class Base
-    attr_accessor :params
+    include ActiveModel::Model
+
+    attr_accessor :course_identifier, :user_id, :raw_event, :lead_provider_from_token, :declaration_date, :declaration_type, :evidence_held
+
+    validates :course_identifier, inclusion: { in: :valid_courses_for_user, message: "The property '#/course_identifier' must be an available course to '#/participant_id'" }
+    validates :declaration_type, inclusion: { in: :valid_declaration_types, message: "The property '#/declaration_type' must be an available for course_identifier '#/course_identifier'" }
+    validates :course_identifier, presence: { message: "The property '#/course_identifier' must be present" }
+    validates :declaration_date, presence: { message: "The property '#/declaration_date' must be present" }
+    validates :declaration_type, presence: { message: "The property '#/declaration_type' must be present" }
+    validates :user, presence: { message: "The participant must be exist" }
+    validates :lead_provider_from_token, presence: { message: "The lead provider must be present" }
```

Comment:
Is this the same? It wasn't supposed to be just a presence check. I'm not seeing where it's checking that the participant is attached to that lead_provider or are we allowing the course check to do that?

Reply:
This does not check that this was the right provider for the user.

This checks that the provider from token exists - cause if it doesn't the creation of our record will fail.

Reply:
And the old check that the lead providers match is still in the same place it used to be in.